packages/opencensus-core/src/metrics/export/metric-producer-manager.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {validateNotNull} from '../../common/validations';
+import {MetricProducer} from '../export/metric-producer';
+
+/**
+ * Keeps a set of MetricProducer that is used by exporters to determine the
+ * metrics that need to be exported.
+ */
+export class MetricProducerManager {
```

Comment:
Should users be able to construct their own, or should only the singleton be used? If users don't need their own, we might not want to export this.

Reply:
> Should users be able to construct their own, or should only the singleton be used?

should only the singleton be used.

> If users don't need their own, we might not want to export this.

Agreed, removed export and added ```MetricProducerManager``` interface type, which is implemented by ```BaseMetricProducerManager```.